Remove deprecated platforms from plugin page

erisu opened the pull request:
### Platforms affected
None

### What does this PR do?
Update plugin list page to remove deprecated platforms.

This PR replaces PR #824. I some how lost PR #824's branch.

### What testing has been done on this change?
Manual Testing

### Checklist
- [ ] [Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database
- [ ] Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected.
- [ ] Added automated test coverage as appropriate for this change.

raphinesse commented:
I've tried to make this a bit more straightforward. Did not test since I've never built the docs before. Might be a matter of taste. Would approve the original PR too :+1:

erisu commented:
@raphinesse I updated the proposal. I added the platform keyword to the sortedPlatforms collection.
Example: ` {keyword: 'cordova-android', icon: 'android', alt: 'Android'},`

The proposed filter `.filter(platform => keywords.has(platform))` did not work because the `keywords` are not identical to `platform`. 

The `keywords` contains a collection of keywords taken from the npm package.json.
For example [cordova-plugin-whitelist](https://www.npmjs.com/package/cordova-plugin-whitelist) you can see the list of keywords at the bottom as `cordova, whitelist, ecosystem:cordova, cordova-android`

filter was updated as so: `.filter(platform => keywords.has(platform.keyword))`
With this, the keyword `cordova-android` will now match platform.keyword `cordova-android`.

raphinesse commented:
@erisu Of course, stupid bug. Thanks for fixing! And thanks for explaining what's actually happening, too :+1:

I probably would have reduced the sorted record format to `{id: 'android', name: 'Android'}` and generated keywords on the fly for filtering, but that's nitpicking :sweat_smile:

erisu commented:
I was also thinking about generating the keyword on the fly for filtering but just couldn't decide what others would really like ha.

erisu commented:
@janpio Let me know if you also have any thoughts, questions, or concerns with this change. At this stage, I think it's good to go unless theres more feedback.

janpio commented on www/static/plugins/supportedplatforms.jsx:
```diff
@@ -1,83 +1,32 @@
+const SupportedPlatforms = createClass({
+    render: function () {
+        // Collection of the npm package keywords.
+        const keywords = new Set(this.props.keywords);
```
(plugins != docs, just in the same repo but actually at https://cordova.apache.org/plugins/)

raphinesse replied:
> IE11 doesn't support the `Set` constructor accepting arguments so this would break there. Will `Set` get transpiled here?

Good catch! I have no idea if there's transpilation. If not, we should revert my whole proposal since IE11 does not support arrow functions either.

erisu replied:
@raphinesse I just uploaded another alternative that doesn't use arrow functions or Set.

As for transpilation, I see a transform process that babelify, in the gulp build task but only has the react preset.

https://github.com/apache/cordova-docs/blob/0c5f1136f56a20582ab58a41757500ea79c71a3b/gulpfile.js#L406

erisu replied:
Out of curiosity, what is the progress of this PR? Is it considered finished or has it been delayed for `babel-preset-env` to be added? 

I believe the scope of this PR had been completed and that the `babel-preset-env` is an additional requirement for the next PR.

raphinesse replied:
@erisu To me, this PR is good to go. I did not test it though and I don't know enough about docs and plugin pages to approve or merge. @janpio might be fit to do that

janpio replied:
Unfortunately I don't know anything about the plugins subproject here.
But as you @raphinesse and @erisu are both ok with the current state, I think this should be safe to merge.

Menardi replied:
@erisu Looks good to me, feel free to merge it (I'm still sorting out my Apache / Github permissions)
